scalableminds/webknossos
# Support remote datasets from s3 compliant data sources

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- [Set up a minio server](https://min.io/docs/minio/linux/index.html)
- Create a bucket in that server and upload a dataset to it (do this via the web console)
- Create an access key on the server
- Explore this dataset with URL: "s3://\<minio-server-ip>:\<minio-server-port>/\<bucket-name>/\<root-key-of-the-dataset>"
- View dataset
- Also explore S3 datasets to ensure they still work

<details>
<summary> Alternative repl testing </summary>

```scala
import java.net.URI;
import com.scalableminds.webknossos.datastore.datavault.S3DataVault;
implicit val ec: scala.concurrent.ExecutionContext = scala.concurrent.ExecutionContext.global;
import com.scalableminds.webknossos.datastore.datavault.VaultPath;
val key = com.scalableminds.webknossos.datastore.storage.S3AccessKeyCredential(name="someKey", accessKeyId="FhXRPMiC9KWMBh9WXF54", secretAccessKey="9eVWQYZckmshjIMsebOBkztMqfHpC5t88bFWOMhc", user="", organization="");
val uri =  new URI("s3://10.191.1.63:8000/somezarr/e75_zarr_local");
val minioVault = new S3DataVault(Some(key), new URI("s3://10.191.1.63:8000/somezarr/e75_zarr_local"));
val vp = new VaultPath(uri, minioVault);
(vp / "color" / "1" / "credentials.json").readBytes().get("")
```

</details>

### TODOs:

### Issues:
- fixes #7432

------
(Please delete unneeded items, merge only when none are left open)
